added github actions to build and release

kbudde opened the pull request:
Hi,

I've added a github action to build and deploy as travis was doing it before.
You must add the following secrets to your repository:
- DOCKERHUB_USERNAME
- DOCKERHUB_TOKEN

#323

kbudde committed:
added github actions to build and release

davideicardi reviewed (approved):
LGTM! Thank you very much!

Maybe we should delete `.github/workflows/maven.yml` ?

I don't have permission to setup github actions variables, I think we need @ekoutanov  .

kbudde commented:
The maven.yml will also run on PRs. Therefore I think it's better to keep it.

mcs commented:
Feel free to do whatever you want with the maven.yml. I just added it to my former PR because there "just" was Travis CI which I could not use for server-side test execution.

mcs commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,90 @@
+      name: push snapshot version
+      run: |
+        echo publishing docker container $app_ver
+        docker push docker.io/obsidiandynamics/kafdrop:$app_ver
```
Couldn't this accidentally overwrite an existing Dockerhub version if one forgets to add "-SNAPSHOT" after a release commit? Afaik this push with a new image id would succeed, while the Github release artifact wouln't change (see line 84).

Maybe it's better to first upload the Github release artifact and only if that succeeds, upload to Dockerhub and Bintray?

kbudde replied:
I like the idea. @mcs  I updated the PR to the order: build, github release, bintray and docker hub.

kbudde committed:
update order of github actions

run github release action first to precent overwriting existing releases.

kbudde commented:
@ekoutanov I would really love to see a release without log4j issues. 
Can you please set up the secrets?

ekoutanov commented:
I'll try to get around to it.

davideicardi commented:
@kbudde @ekoutanov @mcs 
According to #260  Bintray is no longer available. Maybe we should use Github release only?

https://github.com/marketplace/actions/upload-files-to-a-github-release

kbudde committed:
renamed workflows

kbudde committed:
removed bintray upload

ekoutanov commented:
> @kbudde @ekoutanov @mcs According to #260 Bintray is no longer available. Maybe we should use Github release only?
> 
> https://github.com/marketplace/actions/upload-files-to-a-github-release

yeah, Bintray's kaput. We can skip it if the binaries can be uploaded elsewhere.

kbudde committed:
upload github release with artifact

kbudde commented:
@davideicardi , @ekoutanov I fixed it and updated the description.

The github workflow will now create the github release and a tag(!) if the version in pom.xml does end on "-SNAPSHOT". 
The github release has the kafdrop-*VERSION*.jar attached.
See here for an example: https://github.com/kbudde/kafdrop/releases/tag/3.28.0

@ekoutanov only the dockerhub token is needed anymore. 
Bintray was removed. For github the default token is used.

I've also cleaned up the CI files.

Release workflow:
- Update pom.xml (remove `-SNAPSHOT` from version)
- Commit and push to master
- Update pom.xml (increment version + add suffix `-SNAPSHOT` again)
- Commit and push

Each commit on master will push container to dockerhub: `obsidiandynamics/kafdrop:*version*` (e.g. `obsidiandynamics/kafdrop:3.28.0-SNAPSHOT`, `obsidiandynamics/kafdrop:3.28.0`)
If the version is not a snapshot release, another image is pushed:  `obsidiandynamics/kafdrop:latest`.
So latest will always point to last release.

davideicardi commented:
Great work @kbudde ! Thank you!

ekoutanov commented:
Added secrets

davideicardi commented:
@kbudde The new action seems to work without problems!
I just released a new version 3.28.0.
